rust/src/treefile.rs

```diff
@@ -1226,6 +1259,66 @@ etc-group-members:
                 == "table"
         );
     }
+
+    #[test]
+    fn test_split_whitespace_unless_quoted() {
```

Comment:
If we change `split_whitespace_unless_quoted` to be `-> Result<impl Iterator<Item = String>>`, then I'd change this function to be `-> Result<()>` too.